docs/reference/errors-and-warnings/NU1901-NU1904.md

```diff
@@ -0,0 +1,46 @@
+---
+title: NuGet Warnings NU1901, NU1902, NU1903, NU1904
+description: NU1901, NU1902, NU1903, NU1904 Warning codes
+author: zivkan
+ms.author: zivkan
+ms.date: 6/27/2023
+ms.topic: reference
+f1_keywords: 
+  - NU1901
+  - NU1902
+  - NU1903
+  - NU1904
+---
+
+# NuGet Warnings NU1901, NU1902, NU1903, NU1904
+
+```text
+warning NU1902: Package 'NuGet.Protocol' 5.11.2 has a known moderate severity vulnerability, https://github.com/advisories/GHSA-g3q9-xf95-8hp5
+```
+
+The warning code changes depending on the known vulnerability severity level:
+
+|Warning Code|Severity|
+|--|--|
+|NU1901|low|
+|NU1902|moderate|
+|NU1903|high|
+|NU1904|critical|
+
+### Issue
+
+A package restored for your project has a known vulnerability.
+
+### Solution
+
+Upgrading to a newer version of the package is likely to resolve the warning.
+You can check the URL provided by the vulnerability advisory to see what versions of the package have been fixed, or check your configured package source(s) to see what versions of the package are available.
+
+If you do not wish to be notified of vulnerabilities that are less severe than a level you are comfortable with, you can edit the project file and add an MSBuild property `NuGetAuditLevel`, with value set to `low`, `moderate`, `high`, or `severe`.
+For example, `<NuGetAuditLevel>high</NuGetAuditLevel>`.
+
+> [!NOTE]
+> The initial release of NuGetAudit [does not provide a way to suppress specific advisories (URLs)](https://github.com/NuGet/Home/blob/dev/proposed/2022/vulnerabilities-in-restore.md#excluding-advisories).
```

Comment:
I'm guessing you don't want to provide a `NoWarn` as an option?

Reply:
Correct. I actually don't know if `NoWarn` on a `PackageReference` works. I would assume no, that the code generating the warning probably needs to check the PackageReference properties.  But I don't want to encourage customers to suppress the warning for all packages in the project.

Reply:
PackageReference level suppression works for sure. The "logger" has understanding of NoWarn, TreatWarningsAsErrors and the likes. 

If that's your only concern, then you can add it.